theia-rust-docker/Dockerfile

```diff
@@ -0,0 +1,54 @@
+#Can't use node image, because it creates a user that conflicts
+FROM ubuntu:18.04
+
+RUN apt-get update -yq \
+    && apt-get install curl gnupg build-essential gcc g++ gdb make python sudo git -yq 
+
+#allow a user to become root in the container
+ADD sudoers /etc/sudoers
+
+ARG GITHUB_TOKEN
+ARG NODE_OPTIONS="--max-old-space-size=4096"
+ARG NODE_VERSION=10
+
+ADD install_node.sh /root/install_node.sh
+RUN bash /root/install_node.sh --version $NODE_VERSION
+ADD nvm_setup.sh /root/nvm_setup.sh
+
+#install theia
+RUN mkdir /root/theia_rust_extension
+RUN . /root/nvm_setup.sh && cd /root/theia_rust_extension && yarn global add node-gyp
+ADD next.package.json /root/theia_rust_extension/package.json
+RUN git clone https://github.com/eclipse/theia-cpp-extension.git /root/theia-cpp-extension
+RUN cp -R /root/theia-cpp-extension/packages /root/theia_rust_extension/.
```

Comment:
nitpick: `/root/theia_rust_extension` looks like the place where the Theia rust application is build, and so may be better named something like `theia_rust_app` or similar. Could be defined as a ARG or ENV.

Reply:
Created an ENV var